modules/carto/src/sources/types.ts

```diff
@@ -214,7 +214,7 @@ export interface VectorLayer {
   fields: Record<string, string>;
 }
 
-export type TilejsonResult = Tilejson & {accessToken: string};
+export type TilejsonResult = Tilejson & {accessToken: string; matchingColumn?: string};
```

Comment:
Is this the right place for `matchingColumn`? Could we instead have boundaryTableSource return `Promise<TilejsonResult & {matchingColumn: string}>`?

Reply:
Yes, I started with that but it was getting a bit messy - but perhaps it is worth the extra typing though...